pkg/pillar/scripts/generate-self-signed.sh

```diff
@@ -17,26 +17,50 @@ if [ $# != 2 ]; then
 fi
 lifetime=$1
 output_base=$2
+temp_ecc_key=temp.key.pem
 output_key=${output_base}.key.pem
 output_cert=${output_base}.cert.pem
+TPM_ENABLED_FILE="/persist/config/tpm_in_use"
 csr=${output_base}.csr
-dir=$(dirname "$output_key")
+dir=$(dirname "$output_cert")
 if [ ! -d "$dir" ]; then
     echo "Directory does not exist: $dir"
     exit 1
 fi
 
-subject="/C=US/ST=California/L=Santa Clara/O=Zededa, Inc/CN=$(basename "$output_base")"
-openssl ecparam -genkey -name prime256v1 -out "$output_key"
-openssl req -new -sha256 -subj "$subject" -key "$output_key" -out "$csr"
+#The CSR content
+subject="/C=US/ST=California/L=Santa Clara/O=Zededa, Inc/CN=onboard"
+
+if [ -f $TPM_ENABLED_FILE ]; then
+	echo "TPM mode is active, using ECC key from TPM"
+	/opt/zededa/bin/tpmmgr genKey
+	force_pubkey="-force_pubkey /config/tpm.eccpubk.der -keyform DER"
+	csr_key=$temp_ecc_key
+else
+	echo "TPM mode is off, using ECC key from Openssl"
+	force_pubkey=''
+	csr_key=$output_key
+fi
+#Generate a temp ECC key pair, to sign CSR
+openssl ecparam -genkey -name prime256v1 -out $csr_key -noout
+
+#Create CSR
+openssl req -new -sha256 -key $csr_key -subj "$subject" -out $csr
+
+#Create X509 certificate and overwrite public key with TPM key
 # Newer versions require subject - old one fail if it is there
-v=$(openssl version | awk '{print $2}')
+v=`openssl version | awk '{print $2}'`
```

Comment:
This is a regression. shellcheck doesn't like the old backticks.
Please install and run shellcheck manually to see what else needs to be fixed in the modified shell scripts.

Reply:
Addressed.